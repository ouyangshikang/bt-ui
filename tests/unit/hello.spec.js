import { shallowMount } from '@vue/test-utils';
import Hello from '../../packages/hello';

describe('hello.vue', () => {
    it('测试msg参数渲染', () => {
        const msg = 'hello world';
        const wrapper = shallowMount(Hello, {
            propsData: {
                msg
            }
        });
        // 可以读取参数
        console.log(wrapper.props('msg'));
        expect(wrapper.props('msg')).toEqual('hello world');
        expect(wrapper.text()).toMatch(msg);

        // 写入新值再测试
        wrapper.setProps({ msg: '你好世界'});
        expect(wrapper.props('msg')).toEqual('你好世界');
    });

    it('测试dom节点个数', () => {
        const wrapper = shallowMount(Hello);
        expect(wrapper.findAll('.handsome').length).toBe(1);
    });


    // 快照测试
    it('测试组件渲染是否正常', () => {
        const wrapper = shallowMount(Hello, {
            propsData: {
                msg: '你好'
            }
        });
        expect(wrapper).toMatchSnapshot();
    });
});
