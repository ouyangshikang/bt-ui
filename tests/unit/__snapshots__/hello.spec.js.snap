// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hello.vue 测试组件渲染是否正常 1`] = `
<div class="hello">
  <h1>你好</h1> <span class="handsome">你真帅2</span>
</div>
`;
